backend/infrahub/api/query.py

```diff
@@ -120,8 +134,9 @@ async def graphql_query_post(
     ),
     db: InfrahubDatabase = Depends(get_db),
     branch_params: BranchParams = Depends(get_branch_params),
-    _: str = Depends(get_current_user),
+    account_session: AccountSession = Depends(get_current_user),
 ) -> dict:
+    permission_checker: GraphQLQueryPermissionChecker = request.app.state.permission_checker
```

Comment:
I think the permission_checker could be added as a dependency like `get_current_user` or `get_branch_params` above

Reply:
I've updated this to use a dependency now, was on the fence around that as it prevents users from using these names as arguments, but I think the risk is pretty low in this case. Due to circular imports I couldn't place the dependency within the same module as get_current_user and get_branch_params and instead imported it directly from the graphql/depencencies. I think it would be nice with a refactoring of the structure for the web layer in general and clean it up a bit so we can perhaps return to that detail at some later point.